larpandoracontent/LArVertex/LocalAsymmetryFeatureTool.h

```diff
@@ -8,37 +8,24 @@
 #ifndef LAR_LOCAL_ASYMMETRY_FEATURE_TOOL_H
 #define LAR_LOCAL_ASYMMETRY_FEATURE_TOOL_H 1
 
-#include "larpandoracontent/LArVertex/VertexSelectionBaseAlgorithm.h"
+#include "larpandoracontent/LArVertex/AsymmetryFeatureBaseTool.h"
 
 namespace lar_content
 {
 
 /**
  *  @brief  LocalAsymmetryFeatureTool class
  */
-class LocalAsymmetryFeatureTool : public VertexSelectionBaseAlgorithm::VertexFeatureTool
+class LocalAsymmetryFeatureTool : public AsymmetryFeatureBaseTool
 {
 public:
     /**
      *  @brief  Default constructor
      */
     LocalAsymmetryFeatureTool();
 
-    /**
-     *  @brief  Run the tool
-     *
-     *  @param  pAlgorithm address of the calling algorithm
-     *  @param  pVertex address of the vertex
-     *  @param  slidingFitDataListMap map of the sliding fit data lists
-     *
-     *  @return the energy kick feature
-     */
-    void Run(LArMvaHelper::MvaFeatureVector &featureVector, const VertexSelectionBaseAlgorithm *const pAlgorithm, const pandora::Vertex *const pVertex,
-        const VertexSelectionBaseAlgorithm::SlidingFitDataListMap &slidingFitDataListMap, const VertexSelectionBaseAlgorithm::ClusterListMap &,
-        const VertexSelectionBaseAlgorithm::KDTreeMap &, const VertexSelectionBaseAlgorithm::ShowerClusterListMap &, const float, float &);
-
 private:
-    pandora::StatusCode ReadSettings(const pandora::TiXmlHandle xmlHandle);
+    pandora::StatusCode ReadSettings(const pandora::TiXmlHandle xmlHandle) override;
 
     /**
      *  @brief  Get the local asymmetry feature in a given view
```

Comment:
Does it? I'm probably being thick but I can't see which bit needs updating. The method is called `GetAsymmetryForView` because its overriding from the base tool but in this scenario the calculation is specifically for the local asymmetry. The shower cluster list is not used and so is not passed in this overriden version.

Reply:
No, you're right, the third argument is a placeholder for this override use case, I was reading ShowerClusterList as the parameter name, not part of the type. Sorry.

Reply:
Fab! No worries :D